src/main/java/com/study/realworld/global/config/WebSecurityConfig.java

```diff
@@ -1,32 +1,69 @@
 package com.study.realworld.global.config;
 
+import static org.springframework.http.HttpMethod.POST;
+
+import com.study.realworld.security.JwtAuthenticationTokenFilter;
+import com.study.realworld.security.JwtService;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.builders.WebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.factory.PasswordEncoderFactories;
 import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
 
 @Configuration
 public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
 
+    private final JwtService jwtService;
+
+    public WebSecurityConfig(JwtService jwtService) {
+        this.jwtService = jwtService;
+    }
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return PasswordEncoderFactories.createDelegatingPasswordEncoder();
+    }
+
+    @Bean
+    public JwtAuthenticationTokenFilter jwtAuthenticationTokenFilter() {
+        return new JwtAuthenticationTokenFilter(jwtService);
+    }
+
```

Comment:
```suggestion
    @Bean
    public PasswordEncoder passwordEncoder() {
        return PasswordEncoderFactories.createDelegatingPasswordEncoder();
    }

    @Bean
    public JwtAuthenticationTokenFilter jwtAuthenticationTokenFilter(JwtService jwtService) {
        return new JwtAuthenticationTokenFilter(jwtService);
    }

```
이렇게도 가능한걸로 기억하는데 한번 참고해주세요

Reply:
가능한 방법이고 해당 부분에 있어서 더 좋은 방법이라고 생각해요 :)

다만 security에서 `UsernamePasswordAuthenticationFilter`에 대해서 커스텀 filter로 적용하는 부분이 현재 코드에 있어요
```
http
  .addFilterBefore(jwtAuthenticationTokenFilter(jwtService),
    UsernamePasswordAuthenticationFilter.class);
```
그래서 지금처럼 jwtService에 대해서 의존성 주입이 불가피한 상황이기 때문에 다소 코드 변경에 어려움이 있을 것 같아서 조금 더 나은 변경점이 없을까? 하고 변경해보았습니다 :) [새로운 변경점](https://github.com/real-world-study/realworld/pull/15/commits/961f037d159535b8c53a0b54dcc5140b2aecd64f)